tests/drum/test_fit.py

```diff
@@ -319,6 +317,56 @@ def test_transform_fit(
         # Ensure the default type schema is used since we do not disable strict validation
         assert "WARNING: No type schema provided. For transforms, we" in stdout
 
+    @pytest.mark.parametrize(
+        "framework, language",
+        [(SKLEARN_TRANSFORM_NON_NUMERIC, PYTHON), (R_TRANSFORM_NON_NUMERIC, R_FIT),],
+    )
+    @pytest.mark.parametrize("problem", [REGRESSION, BINARY, ANOMALY])
+    @pytest.mark.parametrize("weights", [WEIGHTS_CSV, WEIGHTS_ARGS, None])
+    @pytest.mark.parametrize("strict", [True, False])
+    def test_transform_fit_fails_default(
+        self, resources, framework, language, problem, weights, tmp_path, strict
+    ):
+        """Test that with strict validation non numeric transforms fail, but pass when strict validation is disabled."""
+        custom_model_dir = _create_custom_model_dir(
+            resources, tmp_path, framework, problem, language=framework,
+        )
+
+        input_dataset = resources.datasets(framework, problem)
+
+        weights_cmd, input_dataset, __keep_this_around = self._add_weights_cmd(
+            weights, input_dataset, r_fit=language == R_FIT
+        )
+
+        target_type = TRANSFORM
+
+        cmd = "{} fit --target-type {} --code-dir {} --input {} --verbose {}".format(
+            ArgumentsOptions.MAIN_COMMAND,
+            target_type,
+            custom_model_dir,
+            input_dataset,
+            "" if strict else "--disable-strict-validation",
+        )
+        if problem != ANOMALY:
+            cmd += " --target {}".format(resources.targets(problem))
+
+        if problem in [BINARY, MULTICLASS]:
+            cmd = _cmd_add_class_labels(
+                cmd, resources.class_labels(framework, problem), target_type=target_type
+            )
+
+        cmd += weights_cmd
+
+        if strict:
+            with pytest.raises(AssertionError):
+                _, stdout, _ = _exec_shell_cmd(
+                    cmd, "Failed in {} command line! {}".format(ArgumentsOptions.MAIN_COMMAND, cmd)
```

Comment:
Checking for the actual validation error string here would be idea. Is this what jenkins was giving you trouble with?

Reply:
no, this mostly matches the test above for the cases that pass, but with the pytest.raises.